### ISSUE TITLE
error compiling on ubuntu 16.04 (xenial)

### ISSUE THREAD

**Issue by berndpfrommer** [versatran01/apriltag#17]
I'm getting an error compiling the latest master on Ubuntu 16.04. Not sure what's going on. I have a more recent cmake version installed there. Do you have a Ubuntu 16 box to test on?
```
/home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/src/apriltag_detector.cpp: In member function ‘virtual apriltag_ros::ApriltagVec apr\
iltag_ros::ApriltagDetectorUmich::DetectImpl(const cv::Mat&)’:                                                                                                                
/home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/src/apriltag_detector.cpp:197:5: error: ‘zarray_get’ is not a member of ‘apriltag_ro\
s::umich’                                                                                                                                                                     
     umich::zarray_get(detections.get(), i, &td);                                                                                                                             
     ^                                                                                                                                                                        
/home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/src/apriltag_detector.cpp:197:5: note: suggested alternative:                        
In file included from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/apriltag_umich/common/getopt.h:30:0,                           
                 from /usr/include/unistd.h:874,                                                                                                                              
                 from /usr/include/eigen3/Eigen/src/Core/util/Memory.h:67,                                                                                                    
                 from /usr/include/eigen3/Eigen/Core:299,                                                                                                                     
                 from /usr/include/eigen3/Eigen/Dense:1,                                                                                                                      
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_mit/include/apriltag_mit/AprilTags/GrayModel.h:4,                  
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_mit/include/apriltag_mit/AprilTags/Quad.h:5,                       
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_mit/include/apriltag_mit/AprilTags/TagFamily.h:9,                  
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_mit/include/apriltag_mit/AprilTags/TagDetector.h:8,                
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_mit/include/apriltag_mit/apriltag_mit.h:7,                         
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/include/apriltag_ros/apriltag_detector.h:7,                    
                 from /home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/src/apriltag_detector.cpp:1:                                   
/home/pfrommer/Documents/multicam_calibration/src/tagslam_root/src/apriltag/apriltag_ros/apriltag_umich/common/zarray.h:195:20: note:   ‘zarray_get’                          
 static inline void zarray_get(const zarray_t *za, int idx, void *p)                                                                                                          
                    ^                                                                                  
```

**Comment by versatran01** [comment]
The change I made is that I moved the submodule inside `apriltag_ros`. 
The reason for doing that is the submoduled `apriltag` project is not a catkin package but a pure cmake one. So I moved it inside and add it as a subdirectory so that I can export the include and library of `apriltag` as part of `apriltag_ros`.
This issue was found while Fernando was trying to build a 18.04 docker of apriltag and multicam, I didn't know it would affect 16.04 and have no idea why.
Could it be the cmake version?
